### PR TITLE
VReplication: Add throttler stats

### PR DESCRIPTION
**Pull request by mattlord**
## Description

This pairs with https://github.com/vitessio/vitess/issues/15223 to increase users' ability to observe how the tablet throttler and vreplication interact over time in order to understand the system and explain/debug observed behaviors/results.

## Related Issue(s)

  - Part of: https://github.com/vitessio/vitess/issues/15223

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation: https://github.com/vitessio/website/pull/1692

### CODE REVIEW

File: go/vt/vttablet/tabletmanager/vreplication/vreplicator.go [hunk 1]
```diff
@@ -580,10 +580,18 @@ func (vr *vreplicator) throttlerAppName() string {
...
 
+// updateTimeThrottled updates the time_throttled field in the _vt.vreplication record
+// with a rate limit so that it's only saved in the database at most once per
+// throttleUpdatesRateLimiter.tickerTime.
+// It also increments the throttled count in the stats to keep track of how many
+// times a VReplication workflow, and the specific sub-component, is throttled by the
+// tablet throttler over time.
 func (vr *vreplicator) updateTimeThrottled(appThrottled throttlerapp.Name) error {
+	at := appThrottled.String()
+	vr.stats.ThrottledCounts.Add([]string{"tablet", at}, 1)
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletmanager/vreplication/vreplicator.go:591]: I believe we want to always update stats in a goroutine so as not to make the metrics themselves affect the code's flow (the metric introduces an atomic write).

File: go/vt/vttablet/tabletmanager/vreplication/vreplicator.go [hunk 2]
```diff
@@ -580,10 +580,18 @@ func (vr *vreplicator) throttlerAppName() string {
...
 }
 
+// updateTimeThrottled updates the time_throttled field in the _vt.vreplication record
+// with a rate limit so that it's only saved in the database at most once per
+// throttleUpdatesRateLimiter.tickerTime.
+// It also increments the throttled count in the stats to keep track of how many
+// times a VReplication workflow, and the specific sub-component, is throttled by the
+// tablet throttler over time.
 func (vr *vreplicator) updateTimeThrottled(appThrottled throttlerapp.Name) error {
+	at := appThrottled.String()
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletmanager/vreplication/vreplicator.go:590]: English question: "at" implies a _time_, but here "at" is the app name? Perhaps a different variable name?

**Review by shlomi-noach — COMMENTED** [review]
Looks like a good approach!

File: go/vt/vttablet/tabletmanager/vreplication/vreplicator.go (hunk 1 above)
Comment (mattlord) [reply go/vt/vttablet/tabletmanager/vreplication/vreplicator.go:591]: We don't do that in vreplication today. But I don't disagree with the sentiment.

File: go/vt/vttablet/tabletmanager/vreplication/vreplicator.go (hunk 2 above)
Comment (mattlord) [reply go/vt/vttablet/tabletmanager/vreplication/vreplicator.go:590]: I'll move it to appName

File: go/vt/vttablet/tabletmanager/vreplication/vreplicator.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/vttablet/tabletmanager/vreplication/vreplicator.go]: Yeah, I'm about to get rid of that usage 😛

File: go/vt/vttablet/tabletserver/vstreamer/vstreamer_test.go
Comment (mattlord) [inline go/vt/vttablet/tabletserver/vstreamer/vstreamer_test.go:1]: Unrelated fix for flaky tests seen in the code coverage workflow:
```
--- FAIL: TestMinimalMode (60.00s)
    main_flaky_test.go:79: 
        	Error Trace:	/home/runner/work/vitess/vitess/go/vt/vttablet/tabletserver/vstreamer/main_flaky_test.go:79
        	            				/home/runner/work/vitess/vitess/go/vt/vttablet/tabletserver/vstreamer/vstreamer_test.go:1855
        	Error:      	Received unexpected error:
        	            	could not launch mysql: signal: killed
        	Test:       	TestMinimalMode
```

Comment (mattlord) [inline go/vt/vttablet/tabletserver/vstreamer/vstreamer_test.go:1]: Unrelated fix for flaky tests seen in the code coverage workflow:
```
--- FAIL: TestMinimalMode (60.00s)
    main_flaky_test.go:79: 
        	Error Trace:	/home/runner/work/vitess/vitess/go/vt/vttablet/tabletserver/vstreamer/main_flaky_test.go:79
        	            				/home/runner/work/vitess/vitess/go/vt/vttablet/tabletserver/vstreamer/vstreamer_test.go:1855
        	Error:      	Received unexpected error:
        	            	could not launch mysql: signal: killed
        	Test:       	TestMinimalMode
```

**Review by shlomi-noach — APPROVED** [review]

**Review by rohit-nayak-ps — APPROVED** [review]

**Pull request merged**